frontend/model/state.js

```diff
@@ -126,51 +125,44 @@ const mutations = {
   deleteChatRoomScrollPosition (state, { chatRoomId }) {
     Vue.delete(state.chatRoomScrollPosition, chatRoomId)
   },
-  setChatRoomUnreadSince (state, { chatRoomId, messageHash, createdDate }) {
-    const prevMentions = state.chatRoomUnread[chatRoomId] ? state.chatRoomUnread[chatRoomId].mentions : []
+  setChatRoomReadUntil (state, { chatRoomId, messageHash, createdDate }) {
+    const prevMentions = state.chatRoomUnread[chatRoomId].mentions
     Vue.set(state.chatRoomUnread, chatRoomId, {
-      since: { messageHash, createdDate, deletedDate: null },
+      readUntil: { messageHash, createdDate, deletedDate: null },
       mentions: prevMentions.filter(m => new Date(m.createdDate).getTime() > new Date(createdDate).getTime())
     })
   },
-  setNotificationSettings (state, { chatRoomId, settings }) {
-    if (chatRoomId) {
-      if (!state.notificationSettings[chatRoomId]) {
-        Vue.set(state.notificationSettings, chatRoomId, {})
-      }
-      for (const key in settings) {
-        Vue.set(state.notificationSettings[chatRoomId], key, settings[key])
-      }
-    }
-  },
-  deleteChatRoomUnreadSince (state, { chatRoomId, deletedDate }) {
-    Vue.set(state.chatRoomUnread[chatRoomId], 'since', {
-      ...state.chatRoomUnread[chatRoomId].since,
-      deletedDate
-    })
+  deleteChatRoomReadUntil (state, { chatRoomId, deletedDate }) {
+    Vue.set(state.chatRoomUnread[chatRoomId].readUntil, 'deletedDate', deletedDate)
   },
   addChatRoomUnreadMention (state, { chatRoomId, messageHash, createdDate }) {
     const prevUnread = state.chatRoomUnread[chatRoomId]
-    if (!prevUnread) {
+    if (!prevUnread.readUntil) {
       Vue.set(state.chatRoomUnread, chatRoomId, {
-        since: { messageHash, createdDate, deletedDate: null, fromBeginning: true },
-        mentions: [{ messageHash, createdDate }]
+        readUntil: { messageHash, createdDate, deletedDate: null },
+        mentions: [...prevUnread.mentions, { messageHash, createdDate }]
```

Comment:
Why do you think so?
`state.chatRoomUnread[contractID]` is initiated in the sideEffect of `'gi.contracts/chatroom'` constructor. And it must be before to call any other vuex mutations for this chatroom unread state.

Reply:
Because, before the data stored in Vuex was using other names for keys. And now you're changing those keys. So when clients upgrade to the new code, their existing data is stored on the old keys, not the new keys. You need to properly migrate everything if you change how and where things are stored.

Reply:
I can understand what you mean, and I don't think it would break because of this change. And if the `readUntil` is not set, users will automatically be redirected to the latest message and then `readUntil` will be set to the latest.


Also, we agreed to reset the database [here](https://github.com/okTurtles/group-income/pull/1534#issuecomment-1513972639) after we merge this PR.

Reply:
> Also, we agreed to reset the database here after we merge this PR.

Yes 👍